pkg/processor/build/builder.go

```diff
@@ -1552,6 +1520,72 @@ func (b *Builder) getS3FunctionItemKey() (string, error) {
 	return s3Attributes["s3ItemKey"], nil
 }
 
+func (b *Builder) parseFunctionGitCredentials() (*githttp.BasicAuth, error) {
+	cetGitCredentials := b.options.FunctionConfig.Spec.Build.CodeEntryAttributes["gitCredentials"]
+
+	if cetGitCredentials != nil {
+		type GitCredentials struct {
```

Comment:
imo, instead of having sub-structure of git-credentials - just flatten it.

```
branch string
workdir string
username string
password string
```

these attributes really dont need to be structured, because anyway they are nested under an `interface{}` \ `map[string]string`